konig-core/src/main/java/io/konig/core/pojo/BeanUtil.java

```diff
@@ -228,6 +228,22 @@ public static Value toValue(Object object)  {
 				// Ignore
 			}
 		}
+		if (object instanceof Double) {
+			double doubleValue = (Double)object;
+			if (doubleValue % 1 == 0) {
+			long longValue = (long) doubleValue;
+			if (longValue > Integer.MIN_VALUE && longValue < Integer.MAX_VALUE) {
+			return valueFactory.createLiteral((int) longValue);
+			}
+			if(longValue > Long.MIN_VALUE && longValue < Long.MAX_VALUE) {
```

Comment:
This condition is always true.  There is no point in checking the condition, just return a literal given by the longValue.